### PR TITLE
F#88 single all tactiles pub

### LINKED ISSUE #88: Try single controller approach to tactile state publisher
A single controller class that checks the tactile type and the publishes the right type of msg.

It simplifies the use as we don't have to spawn the right type of tactile controller.

### PR DESCRIPTION
**Pull request by toliver**
Fixes #88

### CODE REVIEW

File: sr_tactile_sensor_controller/include/sr_tactile_sensor_controller/sr_tactile_sensor_controller.hpp [hunk 1]
```diff
@@ -46,14 +46,22 @@
 #include <boost/shared_ptr.hpp>
 #include <ros_ethercat_model/robot_state.hpp>
 
+#include <sr_tactile_sensor_controller/sr_tactile_sensor_publisher.hpp>
+
+//#include <realtime_tools/realtime_publisher.h>
```
Comment (ugocupcic) [inline sr_tactile_sensor_controller/include/sr_tactile_sensor_controller/sr_tactile_sensor_controller.hpp]: why keep them commented out?

File: sr_tactile_sensor_controller/include/sr_tactile_sensor_controller/sr_ubi_tactile_sensor_publisher.hpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ *
+ * Copyright 2015 Shadow Robot Company Ltd.
+ *
+ * @brief  Publishes ubi tactile state
+ *
+ */
+
+/// derived from ImuSensorController  author: Adolfo Rodriguez Tsouroukdissian
+
+#ifndef SR_UBI_TACTILE_SENSOR_CONTROLLER_H
```
Comment (ugocupcic) [inline sr_tactile_sensor_controller/include/sr_tactile_sensor_controller/sr_ubi_tactile_sensor_publisher.hpp]: all the others are using pragma once

File: sr_tactile_sensor_controller/src/sr_ubi_tactile_sensor_publisher.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    // realtime publisher
+    ubi_realtime_pub_ = UbiPublisherPtr(new realtime_tools::RealtimePublisher<sr_robot_msgs::UBI0All>(nh_prefix_, "tactile", 4));
+    midprox_realtime_pub_ = MidProxPublisherPtr(new realtime_tools::RealtimePublisher<sr_robot_msgs::MidProxDataAll>(nh_prefix_, "tactile_mid_prox", 4));
+  }
+
+  void SrUbiTactileSensorPublisher::update(const ros::Time& time, const ros::Duration& period)
+  {
+    bool ubi_published=false;
+    // limit rate of publishing
+    if (publish_rate_ > 0.0 && last_publish_time_ + ros::Duration(1.0/publish_rate_) < time){
```
Comment (ugocupcic) [inline sr_tactile_sensor_controller/src/sr_ubi_tactile_sensor_publisher.cpp]: bad formatting?

should be 

```
if
{
```

**Comment by ugocupcic** [comment]
can you fix circle ci (just ignore the external depencies for now, but fix the others). Happy to merge once circle ci is happy and that you've addressed the comment about commented out inclusion of the headers.

**Comment by toliver** [comment]
Many of the lint errors belong to files that are not part of this PR (or the external dependencies). Should I fix those as well, or create an issue?

**Pull request merged**